packages/third-parties/src/google/gtm.tsx

```diff
@@ -43,11 +50,13 @@ export function GoogleTagManager(props: GTMParams) {
         ${dataLayer ? `w[l].push(${JSON.stringify(dataLayer)})` : ''}
       })(window,'${dataLayerName}');`,
```

Comment:
[Google docs](https://developers.google.com/tag-platform/security/guides/csp) have a modified version of scripts when using a CSP, that seems to pass the nonce as an attribute. This might need to change

Reply:
@Vinnl Do you have the bandwidth to update your PR?

Reply:
Thanks for asking, but not really - I don't use GTM myself, so I'm not sure what should be needed here, and it doesn't look like this PR is going to get looked at anyway...

Reply:
Thanks for the reply. Basically need a small fix to the script according to @Stuj1's suggestion. Do you want to give it a try or I can raise another PR to fix that?

Reply:
Happy to merge a PR into mine, or for you to create a separate PR - whatever works for you!

Reply:
would be great if this can be used soon!

Reply:
Hi missed this comment originally. I am not able to add anything to this PR but I made the necessary fixes here:

https://github.com/zhawtof/next.js/tree/pr/61820

Reply:
@zhawtof Maybe create a new PR with those changes? Then you can add own the changes and review feedback, and add a proper description describing the added changes and why they are needed?

Reply:
Done @Vinnl. Included your commit.

https://github.com/vercel/next.js/pull/69931